src/operator/exhaustMap.ts

```diff
@@ -6,35 +6,33 @@ import {errorObject} from '../util/errorObject';
 import {OuterSubscriber} from '../OuterSubscriber';
 import {subscribeToResult} from '../util/subscribeToResult';
 
-export function exhaustMap<T, R, R2>(project: (value: T, index: number) => Observable<R>,
-                                     resultSelector?: (outerValue: T,
-                                                       innerValue: R,
-                                                       outerIndex: number,
-                                                       innerIndex: number) => R2): Observable<R> {
+export function exhaustMap<T, R, TResult>(project: (value: T, index: number) => Observable<R>,
```

Comment:
Fair enough, in C# land generally generic types are just prefixed with `T`
to distinguish them, and you'll almost never see a class with TMyClass.
Worth asking the question. :)

Thanks.

On Thu, Dec 17, 2015 at 11:59 AM, Luís Gabriel Lima <
notifications@github.com> wrote:

> In src/operator/exhaustMap.ts
> <https://github.com/ReactiveX/RxJS/pull/1077#discussion_r47930605>:
>
> > @@ -6,35 +6,33 @@ import {errorObject} from '../util/errorObject';
> >  import {OuterSubscriber} from '../OuterSubscriber';
> >  import {subscribeToResult} from '../util/subscribeToResult';
> >
> > -export function exhaustMap<T, R, R2>(project: (value: T, index: number) => Observable<R>,
> > -                                     resultSelector?: (outerValue: T,
> > -                                                       innerValue: R,
> > -                                                       outerIndex: number,
> > -                                                       innerIndex: number) => R2): Observable<R> {
> > +export function exhaustMap<T, R, TResult>(project: (value: T, index: number) => Observable<R>,
>
> It's not clear looking at a variable declaration, for example, if TResult
> is a type parameter or a class/interface. Also, it's not consistent with
> the pattern we're using to name type parameters in this codebase as well as
> in RxJava.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/ReactiveX/RxJS/pull/1077/files#r47930605>.
>